linera-service/tests/linera_net_tests.rs

```diff
@@ -4327,6 +4327,95 @@ async fn test_end_to_end_faucet_with_long_chains(config: impl LineraNetConfig) -
+    let _guard = INTEGRATION_TEST_GUARD.lock().await;
+    tracing::info!("Starting test {}", test_name!());
+
+    let (mut net, client1) = config.instantiate().await?;
+
+    let chain1 = client1.load_wallet()?.default_chain().unwrap();
+    let balance1 = client1.local_balance(Account::chain(chain1)).await?;
+
+    // Start faucet with small batch size for testing
+    let mut faucet_service = client1
+        .run_faucet(None, chain1, Amount::from_tokens(2))
+        .await?;
+    let faucet = faucet_service.instance();
+
+    // Test batch processing by creating multiple concurrent requests
+    const NUM_REQUESTS: usize = 20;
+    let mut handles = Vec::new();
+
+    for i in 0..NUM_REQUESTS {
+        let faucet_clone = faucet.clone();
+        handles.push(async move {
+            let owner = AccountOwner::from(
+                AccountSecretKey::Secp256k1(Secp256k1SecretKey::generate()).public(),
+            );
+            tracing::info!("Request {} claiming chain for owner: {}", i, owner);
+            faucet_clone.claim(&owner).await
+        });
+    }
+
+    // Wait for all requests to complete
+    let chain_descriptions = future::try_join_all(handles).await?;
+
+    // Verify all chains were created successfully
+    assert_eq!(chain_descriptions.len(), NUM_REQUESTS);
+
+    // Verify all chains are unique
+    let mut chain_ids = std::collections::HashSet::new();
+    for desc in &chain_descriptions {
+        assert!(
+            chain_ids.insert(desc.id()),
+            "Duplicate chain ID: {}",
+            desc.id()
+        );
+    }
+
+    // Test duplicate request handling - should return existing chain
+    let owner =
+        AccountOwner::from(AccountSecretKey::Secp256k1(Secp256k1SecretKey::generate()).public());
+    let first_claim = faucet.claim(&owner).await?;
+    let second_claim = faucet.claim(&owner).await?;
+    assert_eq!(
+        first_claim.id(),
+        second_claim.id(),
+        "Duplicate request should return same chain"
+    );
+
+    faucet_service.ensure_is_running()?;
+    faucet_service.terminate().await?;
+
+    // Verify balance was decremented appropriately (NUM_REQUESTS * 2 tokens + fees)
+    let final_balance = client1.query_balance(Account::chain(chain1)).await?;
+    let expected_transfer = Amount::from_tokens((NUM_REQUESTS * 2) as u128);
+    assert!(final_balance <= balance1 - expected_transfer);
+
+    // Verify that fewer than NUM_REQUESTS blocks were created, i.e. some of them were batched.
+    let port = get_node_port().await;
+    let service = client1.run_node_service(port, ProcessInbox::Skip).await?;
+    let query =
+        format!("query {{ chain(chainId:\"{chain1}\") {{ tipState {{ nextBlockHeight }} }} }}");
+    let response = service.query_node(query).await?;
+    let height = response["chain"]["tipState"]["nextBlockHeight"]
+        .as_u64()
+        .unwrap();
+    assert!(height < NUM_REQUESTS as u64);
```

Comment:
This doesn't really verify that we can batch up tp `max_batch_size` operations per block. Is it too low-level of the test to write?

Reply:
I see it's a different test. So this test is a bit hand-wave'y. Do you think we need it given you have other tests added?

Reply:
True, we could remove the test. Not sure what's best.